src/include/homestore/blk.h

```diff
@@ -163,27 +218,40 @@ std::basic_ostream< charT, traits >& operator<<(std::basic_ostream< charT, trait
     return outStream;
 }
 
-/* Hints for various allocators */
+template < typename charT, typename traits >
+std::basic_ostream< charT, traits >& operator<<(std::basic_ostream< charT, traits >& outStream, BlkId const& blk) {
+    return stream_op< charT, traits, BlkId >(outStream, blk);
+}
+
+template < typename charT, typename traits >
+std::basic_ostream< charT, traits >& operator<<(std::basic_ostream< charT, traits >& outStream, MultiBlkId const& blk) {
+    return stream_op< charT, traits, MultiBlkId >(outStream, blk);
+}
+
+///////////////////// Other common Blkd definitions /////////////////////
+VENUM(BlkAllocStatus, uint32_t,
+      BLK_ALLOC_NONE = 0,        // No Action taken
+      SUCCESS = 1ul << 0,        // Success
+      FAILED = 1ul << 1,         // Failed to alloc/free
+      REQ_MORE = 1ul << 2,       // Indicate that we need more
+      SPACE_FULL = 1ul << 3,     // Space is full
+      INVALID_DEV = 1ul << 4,    // Invalid Device provided for alloc
+      PARTIAL = 1ul << 5,        // In case of multiple blks, only partial is alloced/freed
+      INVALID_THREAD = 1ul << 6, // Not possible to alloc in this thread
+      INVALID_INPUT = 1ul << 7,  // Invalid input
+      TOO_MANY_PIECES = 1ul << 8 // Allocation results in more pieces than passed on
+);
+
 struct blk_alloc_hints {
-    blk_alloc_hints() :
-            desired_temp{0},
-            dev_id_hint{INVALID_DEV_ID},
-            can_look_for_other_chunk{true},
-            is_contiguous{false},
-            multiplier{1},
-            max_blks_per_entry{BlkId::max_blks_in_op()},
-            stream_info{(uintptr_t) nullptr} {}
-
-    blk_temp_t desired_temp;       // Temperature hint for the device
-    uint32_t dev_id_hint;          // which physical device to pick (hint if any) -1 for don't care
-    bool can_look_for_other_chunk; // If alloc on device not available can I pick other device
-    bool is_contiguous;
-    uint32_t multiplier;           // blks allocated in a blkid should be a multiple of multiplier
-    uint32_t max_blks_per_entry;   // Number of blks on every entry
-    uintptr_t stream_info;
-#ifdef _PRERELEASE
-    bool error_simulate = false; // can error simulate happen
-#endif
+    blk_temp_t desired_temp{0};                  // Temperature hint for the device
+    std::optional< uint32_t > pdev_id_hint;      // which physical device to pick (hint if any) -1 for don't care
+    std::optional< chunk_num_t > chunk_id_hint;  // any specific chunk id to pick for this allocation
```

Comment:
I saw you are already adding chunk_id_hint do you want to also handle it in `VirtualDev::do_alloc_blk`, or I can do it with my PR too, just it might cause some merge conflict to your PR if I merge first, which is a small issue.

Reply:
I think we can handle the chunk_id hints for blk allocation as part of another PR.